File: 1Urte desberdinetako materiala/Ethazi 25-26/multi_intefaz/src/main/java/model/IdazleData.java

package model;

import java.util.List;


public class IdazleData {
    private int id;
    private String izena;
    //private String idazle_izena;
    private List<Liburu> liburuak;

    public IdazleData(int id, String izena,  List<Liburu> liburuak) {
        this.id = id;
        this.izena = izena;
       // this.idazle_izena = idazle_izena;
        this.liburuak = liburuak;
    }

//    public IdazleData() {
//    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getIzena() {
        return izena;
    }

    public void setIzena(String izena) {
        this.izena = izena;
    }

//    public String getIdazle_izena() {
//        return idazle_izena;
//    }
//
//    public void setIdazle_izena(String idazle_izena) {
//        this.idazle_izena = idazle_izena;
//    }

    public List<Liburu> getLiburuak() {
        return liburuak;
    }

    public void setLiburuak(List<Liburu> liburuak) {
        this.liburuak = liburuak;
    }

    
    // Opcional: para mostrar el nombre en el ComboBox
    @Override
    public String toString() {
        return izena ;
    }
    
}

File: PMD 25-26/Erronka/app/src/main/java/com/example/erronka/model/IdazleData.kt
package com.example.erronka.model

/**class IdazleData // this.idazle_izena = idazle_izena;
    (//    public IdazleData() {
    //    }
    var id: Int, var izena: String?, //    public String getIdazle_izena() {
    //        return idazle_izena;
    //    }
    //
    //    public void setIdazle_izena(String idazle_izena) {
    //        this.idazle_izena = idazle_izena;
    //    }
    //private String idazle_izena;
    var liburuak: MutableList<Liburu?>?
) {
    // Opcional: para mostrar el nombre en el ComboBox
    override fun toString(): String {
        return izena!!
    }
}**/

class IdazleData(
    private var id: Int,
    private var izena: String,
    // List eta MutableList-en arteko desberdintasuna da
    // List soilik balio duela datuak irakurteko eta MutableList
    // add erabiltzea usten du eraz alda daiteke
    private var liburuak: MutableList<Liburu> = mutableListOf()
) {

    // Getters y Setters
    fun getId(): Int = id
    fun setId(id: Int) { this.id = id }

    fun getIzena(): String = izena
    fun setIzena(izena: String) { this.izena = izena }

    // Getter público: devuelve MutableList para poder usar add()
    fun getLiburuak(): MutableList<Liburu> = liburuak

    // Setter público: recibe List o MutableList y lo convierte internamente a MutableList
    fun setLiburuak(list: List<Liburu>) {
        liburuak = list.toMutableList()
    }

    // Liburua gehitzeko metodoa
    fun addLiburu(liburu: Liburu) {
        liburuak.add(liburu)
    }

    // toString para mostrar solo el nombre
    override fun toString(): String = izena
}